src/ussd_optout_rapidpro.js

```diff
@@ -1,29 +1,333 @@
+                        return self.states.create("state_opt_out");
+                    } else if(self.contact_in_group(contact, self.im.config.clinic_group_ids)){
+                        return self.states.create("state_opt_out");
+                    } else {
+                        return self.states.create("state_check_previous_optout");
+                    }
+                }).catch(function(e) {
+                    // Go to error state after 3 failed HTTP requests
+                    opts.http_error_count = _.get(opts, "http_error_count", 0) + 1;
+                    if(opts.http_error_count === 3) {
+                        self.im.log.error(e.message);
+                        return self.states.create("__error__");
+                    }
+                    return self.states.create("state_start", opts);
+                });
+        });
+
+        self.states.add("state_opt_out", function(name) {
+            return new MenuState(name, {
+                question: $("Hello mom! Do you want to stop getting MomConnect (MC) messages?"),
+                error: $(
+                    "Sorry, please reply with the number next to your answer. " +
+                    "Do you want to stop getting MomConnect (MC) messages?"
+                ),
+                choices: [
+                    new Choice("state_optout_reason", $("Yes")),
+                    new Choice("state_no_optout", $("No"))
+                ],
+            });
+        });
+
+        self.states.add("state_no_optout", function(name) {
+            return new EndState(name, {
+                next: "state_start",
+                text: $(
+                    "Thanks! MomConnect will continue to send you helpful messages and process your " +
+                    "personal info. Have a lovely day!"
+                )
+            });
+        });
+
+        self.add("state_optout_reason", function(name) {
+            var question = "We'll stop sending msgs. Why do you want to stop your MC msgs?";
+            return new PaginatedChoiceState(name, {
+                question: $(question),
+                error: $(question),
+                accept_labels: true,
+                options_per_page: null,
+                characters_per_page: 160,
                 choices: [
                     new Choice("miscarriage", $("Miscarriage")),
-                    new Choice("stillbirth", $("Baby was stillborn")),
-                    new Choice("babyloss", $("Baby died")),
-                    new Choice("not_useful", $("Messages not useful")),
-                    new Choice("other", $("Other"))
+                    new Choice("stillborn", $("Baby was stillborn")),
+                    new Choice("babyloss", $("Baby passed away")),
+                    new Choice("not helpful", $("Msgs aren't helpful")),
+                    new Choice("other", $("Other")),
+                    new Choice("not stated", $("I prefer not to say"))
```

Comment:
I think we should stick with the existing reasons that we're already using in the event store, so that we don't have to translate between the two when storing this, and when submitting to DHIS2: https://github.com/praekeltfoundation/ndoh-hub/blob/develop/eventstore/models.py#L35-L41

Reply:
thanks. Was not sure what to use

Reply:
So I could change the choices to this? 
```new Choice("Miscarriage", $("Miscarriage")),
                    new Choice("Stillborn", $("Baby was stillborn")),
                    new Choice("Babyloss", $("Baby passed away")),
                    new Choice("Not useful", $("Msgs aren't helpful")),
                    new Choice("Other", $("Other")),
                    new Choice("Unknown", $("I prefer not to say"))```

Reply:
I think we need to stick to the machine versions, `not_useful`, `babyloss`, etc, rather than the human-readable versions.